homeassistant/components/sensor/wunderground.py

```diff
@@ -639,10 +639,7 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     for variable in config[CONF_MONITORED_CONDITIONS]:
         sensors.append(WUndergroundSensor(rest, variable))
 
-    try:
-        rest.update()
-    except ValueError as err:
-        _LOGGER.error("Received error from WUnderground: %s", err)
+    if not rest.update():
```

Comment:
I like the intention here, but we shouldn't rely on the return value since this is a throttled method. We don't want this to seem like a safe pattern to reuse elsewhere. Instead write it like this:
```python
rest.update()
if not rest.data:
    return False
```

Reply:
And what to remove update completely from setup ?

Reply:
We should still check to see if we can connect in setup, but what you can do is instead of `return False`, you can `raise PlatformNotReady` so Home Assistant knows it can retry later.

Example:
https://github.com/home-assistant/home-assistant/blob/6da08deabf0ede821bc8c0e3cb9b470608d7ea7a/homeassistant/components/switch/rainbird.py#L50-L52